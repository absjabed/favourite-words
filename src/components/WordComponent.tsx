import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome'
import Colors from '../constants/Colors/Colors';
import AppText from './Text';

type Props = {
    addToFavourites: (item: any) => void;
    wordItem: any;
    index: number;
  };

const WordComponent = ({addToFavourites, index, wordItem}: Props) => {
  return (
    <View style={{ ...styles.cardShadow, flex:1, flexDirection:'row', backgroundColor:'#fff', padding:10, borderRadius:10, marginBottom: 10}} key={index} >
        <View style={{flexDirection:'column', width:'100%', justifyContent:'center', alignItems:'center'}}>
            <Text style={{backgroundColor: Colors.primary, borderRadius:7, justifyContent:'flex-start',  fontFamily:'Ubuntu-Regular', padding:5, color: Colors.white, alignSelf:'center'}}>{wordItem.type}</Text>
            <Icon onPress={()=> addToFavourites(wordItem)} style={{position:'absolute', right:0, top: 0}} name="heart-o" size={20} color={Colors.secondaryGradientEnd} />
            <AppText style={{fontSize:12, padding:2, textAlign:'center', fontFamily:'Ubuntu-Bold', color: Colors.black}} >{wordItem.definition}</AppText>
            {wordItem.image_url ? (<View>
                <Image style={styles.image}  source={{uri:wordItem.image_url}} />
                </View>) : <></>}
            <AppText style={{fontSize:12, padding:2, textAlign:'center', fontFamily:'Ubuntu-MediumItalic', color: Colors.darkGray}} >"{wordItem.example}"</AppText>
        </View>
    </View>
  );
};

export default WordComponent;

const styles = StyleSheet.create({
  container: {
    display:'flex', 
    flexDirection:'row', 
    width:'100%', 
    justifyContent:'center', 
    alignContent:'center', 
    alignItems:'center'
  },
  cardShadow:{
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.20,
    shadowRadius: 1.41,
    elevation: 3,

    borderWidth:0.3,
    borderRadius:10,
    borderColor:'green'
  },
  image: {
    width: 90,
    height: 90,
    borderRadius: 10,
    borderWidth: 4,
    borderColor: "white",
    alignSelf:'center',
  },
});
