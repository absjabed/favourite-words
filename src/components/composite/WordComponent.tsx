import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome'
import Colors from '../../constants/Colors/Colors';
import AppText from '../Text';

type Props = {
    addToFavourites: (item: any) => void;
    wordItem: any;
    index: number;
  };

const WordComponent = ({addToFavourites, index, wordItem}: Props) => {
  return (
    <View style={{ ...styles.cardShadow, ...styles.container}} key={index} >
        <View style={{flexDirection:'column', width:'100%', justifyContent:'center', alignItems:'center'}}>
            <Text style={{backgroundColor: Colors.primary, borderRadius:7, justifyContent:'flex-start',  fontFamily:'Ubuntu-Regular', padding:5, color: Colors.white, alignSelf:'center'}}>{wordItem.type}</Text>
            <Icon onPress={()=> addToFavourites(wordItem)} style={{position:'absolute', right:0, top: 0}} name="heart" size={20} color={Colors.secondaryGradientEnd} />
            <AppText style={{fontSize:12, padding:2, textAlign:'center', fontFamily:'Ubuntu-Regular', paddingTop:5, color: Colors.white}} >{wordItem.definition}</AppText>
            {wordItem.image_url ? (<View>
                <Image style={styles.image}  source={{uri:wordItem.image_url}} />
                </View>) : <></>}
            <AppText style={{fontSize:12, padding:2, textAlign:'center', fontFamily:'Ubuntu-MediumItalic', color: Colors.grey}} >"{wordItem.example}"</AppText>
        </View>
    </View>
  );
};

export default WordComponent;

const styles = StyleSheet.create({
  container: {
    flex:1, 
    flexDirection:'row', 
    backgroundColor: Colors.cardBackground, 
    padding:10, 
    borderRadius:10, 
    marginBottom: 10
  },
  cardShadow:{
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.20,
    shadowRadius: 1.41,
    elevation: 3,

    borderWidth:0.3,
    borderRadius:10,
    borderColor: Colors.grey
  },
  image: {
    width: 90,
    height: 90,
    borderRadius: 10,
    borderWidth: 1,
    margin: 5,
    borderColor: "white",
    alignSelf:'center',
  },
});
