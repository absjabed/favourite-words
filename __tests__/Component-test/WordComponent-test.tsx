import 'react-native';
import React from 'react';
import WordComponent from '../../src/components/composite/WordComponent';
import renderer, { create } from 'react-test-renderer';

const mockAddToFavouriteFunc = () => console.log('added to favoutites'),
mockWordItem = {
      "definition": "a nocturnal bird of prey with large eyes, a facial disc, a hooked beak, and typically a loud hooting call.", 
      "emoji": "ðŸ¦‰", 
      "example": "I love reaching out into that absolute silence, when you can hear the owl or the wind.", 
      "image_url": "https://media.owlbot.info/dictionary/images/owl.jpg.400x400_q85_box-403,83,960,640_crop_detail.jpg", 
      "type": "noun", 
      "word": "owl"
    },
mockIndex = 0;

it('WordComponent renders correctly', () => {
  renderer.create(<WordComponent wordItem={mockWordItem} index={mockIndex} addToFavourites={mockAddToFavouriteFunc} />);
});

test('WordComponent snapshot', () => {
  const tree = create(<WordComponent wordItem={mockWordItem} index={mockIndex} addToFavourites={mockAddToFavouriteFunc} />);
  expect(tree).toMatchSnapshot();
});
