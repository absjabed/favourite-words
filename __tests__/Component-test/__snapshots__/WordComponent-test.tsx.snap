// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordComponent snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#22272e",
      "borderColor": "#acacac",
      "borderRadius": 10,
      "borderWidth": 0.3,
      "elevation": 3,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#555CC4",
          "borderRadius": 7,
          "color": "#ffffff",
          "fontFamily": "Ubuntu-Regular",
          "justifyContent": "flex-start",
          "padding": 5,
        }
      }
    >
      noun
    </Text>
    <Icon
      color="#FF1358"
      name="heart-o"
      onPress={[Function]}
      size={20}
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Ubuntu-Regular",
          "fontSize": 12,
          "padding": 2,
          "paddingTop": 5,
          "textAlign": "center",
        }
      }
    >
      a nocturnal bird of prey with large eyes, a facial disc, a hooked beak, and typically a loud hooting call.
    </Text>
    <View>
      <Image
        source={
          Object {
            "uri": "https://media.owlbot.info/dictionary/images/owl.jpg.400x400_q85_box-403,83,960,640_crop_detail.jpg",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 90,
            "margin": 5,
            "width": 90,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#acacac",
          "fontFamily": "Ubuntu-MediumItalic",
          "fontSize": 12,
          "padding": 2,
          "textAlign": "center",
        }
      }
    >
      "
      I love reaching out into that absolute silence, when you can hear the owl or the wind.
      "
    </Text>
  </View>
</View>
`;
